alarm/nialarm.yml

```diff
@@ -0,0 +1,686 @@
+        description: The type of resource associated with a given alarm instance
+        type: string
+        example: Tag
+  AlarmNote:
+    title: Alarm Note
+    description: Information about a particular alarm instance, such as a description of an investigation into an alarm's root cause.
+    required:
+      - note
+    properties:
+      note:
+        description: Information about a particular alarm instance, such as a description of an investigation into an alarm's root cause.
+        type: string
+        example: Temperature sensor was faulty
+      createdAt:
+        description: The time at which the note was created
+        type: string
+        example: '2019-01-14T23:15:53.7270539Z'
+      user:
+        description: The user who created the note
+        type: string
+        readOnly: true
+        example: Philip
+  AlarmTransitionType:
+    title: Alarm Transition Type
+    type: string
+    enum: [SET, CLEAR]
+  AlarmTransition:
+    title: Alarm Transition
+    description: A transition within a given alarm instance. Transitions can record changes to an alarm's *clear* field, as well as an alarm increasing or decreasing in severity.
+    type: object
+    required:
+      - transitionType
+    properties:
+      transitionType:
+        $ref: '#/definitions/AlarmTransitionType'
+      occurredAt:
+        description: The time at which the transition occurred.
+        type: string
+        example: '2019-01-14T23:15:53.7270539Z'
+      severityLevel:
+        description: The severity of the transition. Valid values for CLEAR transitions are [-1,-1]. Valid values for SET transitions are [1, 4].
+        type: integer
+        example: 2
+        default: 2
+        minimum: -1
+        maximum: 4
+      notificationStrategyIds:
+        description: The IDs of the notification strategies which should be triggered if this request results in an alarm instance being created or transitioning to a new highest severity
+        type: array
+        items:
+          type: string
+        example: []
+      condition:
+        description: The condition associated with the transition
+        type: string
+        example: Greater than 90
+      shortText:
+        description: Optional additional description of the condition
+        type: string
+        example: Low disk space
+      detailText:
+        description: Optional additional description of the condition
+        type: string
+        example: Disk usage on CRIO1 is 95% (lower than configured threshold of 90%)
+      keywords:
+        description: Keywords associated with a transition
+        type: array
+        items:
+          type: string
+        example:
+          - keyword1
+          - keyword2
+      properties:
+        description: Key-value-pair metadata associated with a transition
+        type: object
+        additionalProperties:
+          type: string
+        example:
+          key1: value1
+  SubQuery:
+    title: Nested Query Body
```

Comment:
It can make it a little confusing to have the title too different from the model name. For example, the codegen will ignore the title and generate a model named SubQuery.

Also, should the model have a description? Several of its properties are missing descriptions as well.

Reply:
Changed the title and added missing descriptions.